chord_metadata_service/restapi/models.py

```diff
@@ -1,4 +1,92 @@
+from abc import abstractmethod
+from django.apps import apps
+from django.db import models
+from django.db.models import Q, QuerySet
+from django.core.exceptions import ValidationError, ObjectDoesNotExist
+from jsonschema import Draft7Validator
+from typing import Tuple, List
+from chord_metadata_service.logger import logger
+
+
 class IndexableMixin:
     @property
     def index_id(self):
         return f"{self.__class__.__name__}|{self.__str__()}"
+
+
+class SchemaType(models.TextChoices):
+    PHENOPACKET = "PHENOPACKET", "Phenopacket"
+    BIOSAMPLE = "BIOSAMPLE", "Biosample"
+    INDIVIDUAL = "INDIVIDUAL", "Individual"
+
+
+class BaseExtraProperties(models.Model):
+    extra_properties = models.JSONField()
+
+    class Meta:
+        abstract = True
+
+    @property
+    @abstractmethod
+    def schema_type(self) -> SchemaType:
```

Comment:
add `# pragma: no cover` after the `:` on this function since it's an abstract method, to exclude it from coverage calculation